### DISCUSSION TITLE
Meaning of wth, rth, c80, c280 in benchmarks

### DISCUSSION THREAD

**Discussion by sriram-kondakindi** [emmett-framework/granian#206] [Q&A] [answered]
Hi everyone,

Can anyone let me know the meaning of wth, rth, c80, c280 etc in the benchmarks page

![Screenshot_2024-02-09-02-45-23-85_320a9a695de7cdce83ed5281148d6f19.jpg](https://github.com/emmett-framework/granian/assets/32274323/ad99683b-6769-4098-945f-bf69a4b11c44)

Thanks

**Comment by gi0baro** [comment, marked as answer]
`wth` and `rth` are the *workers* and *runtime* threading modes [as described in README](https://github.com/emmett-framework/granian?tab=readme-ov-file#threading-mode).

`c{number}` is the best resulting connection concurrency used in `wrk` command. If you take a look at [the banchmarks code](https://github.com/emmett-framework/granian/blob/master/benchmarks/benchmarks.py#L12) you can see every run is performed with different concurrencies and [the final summary](https://github.com/emmett-framework/granian/blob/master/benchmarks/_template.md?plain=1#L15) picks the best one.

**Reply by sriram-kondakindi** [reply]
Thank you for the clarification
